tests/io/test_multi_table.py

```diff
@@ -0,0 +1,241 @@
+        # there is already a function to do one of these two join operations which is match_table_to_element()
+        # is too verbose/complex we need to adjust the internals to make it smoother
+        # # use case example 1
+        # # sorting the shapes to match the order of the table
+        # alternatively, we can have a helper function (join, and simpler ones "match_table_to_element()"
+        # "match_element_to_table()", "match_annotations_order(...)", "mathc_reference_eleemnt_order??(...)")
+        # sdata["visium0"][SpatialData.get_instance_key_column(sdata.table['visium0'])]
+        # assert ...
+        # # use case example 2
+        # # sorting the table to match the order of the shapes
+        # sdata.table.obs.set_index(keys=["__instance_id__"])
+        # sdata.table.obs[sdata["visium0"]]
+        # assert ...
+
+    def test_paired_elements_tables(self, tmp_path: str):
+        tmpdir = Path(tmp_path) / "tmp.zarr"
+        table = _get_table(region="poly")
+        table2 = _get_table(region="multipoly")
+        table3 = _get_table(region="non_existing")
+        with pytest.warns(UserWarning, match="The table is annotating elements not present in the SpatialData object"):
+            SpatialData(
+                shapes={"poly": test_shapes["poly"], "multipoly": test_shapes["multipoly"]},
+                table={"poly_annotate": table, "multipoly_annotate": table3},
+            )
+        test_sdata = SpatialData(
+            shapes={"poly": test_shapes["poly"], "multipoly": test_shapes["multipoly"]},
+            table={"poly_annotate": table, "multipoly_annotate": table2},
+        )
+        test_sdata.write(tmpdir)
+        test_sdata = SpatialData.read(tmpdir)
+        assert len(test_sdata.tables) == 2
+
+    def test_single_table_multiple_elements(self, tmp_path: str):
+        tmpdir = Path(tmp_path) / "tmp.zarr"
+        table = _get_table(region=["poly", "multipoly"])
+        subset = table[table.obs.region == "multipoly"]
+        with pytest.raises(ValueError, match="Regions in"):
+            TableModel().validate(subset)
+
+        test_sdata = SpatialData(
+            shapes={
+                "poly": test_shapes["poly"],
+                "multipoly": test_shapes["multipoly"],
+            },
+            table=table,
+        )
+        test_sdata.write(tmpdir)
+        SpatialData.read(tmpdir)
+
+        # # use case example 1
+        # # sorting the shapes visium0 to match the order of the table
+        # sdata["visium0"][sdata.table.obs["__instance_id__"][sdata.table.obs["__spatial_element__"] == "visium0"]]
+        # assert ...
+        # # use case example 2
+        # # subsetting and sorting the table to match the order of the shapes visium0
+        # sub_table = sdata.table[sdata.table.obs["__spatial_element"] == "visium0"]
+        # sub_table.set_index(keys=["__instance_id__"])
+        # sub_table.obs[sdata["visium0"]]
+        # assert ...
```

Comment:
As commented before, we need to be careful not to remove this comment as the tests that it describes are super important, but we can address them in a follow up PR.